api/swagger.yml

```diff
@@ -2523,7 +2554,98 @@ paths:
           description: too many requests
         default:
           $ref: "#/components/responses/ServerError"
-
+  /auth/external/principals:
```

Comment:
I'm not sure what this API returns.
a list of external principals configured for the current installation?

Reply:
I used the exact pattern we use for groups, roles, users. 
depends on the HTTP verb, one is a list the other will create attachment.